packages/core/src/components/Link/Link.test.tsx

```diff
@@ -39,4 +39,31 @@ describe('<Link />', () => {
     });
     expect(getByText(testString)).toBeInTheDocument();
   });
+
+  describe('isExternalUri', () => {
+    it.each([
+      [true, 'http://'],
+      [true, 'https://'],
+      [true, 'https://some-host'],
+      [true, 'https://some-host/path#fragment'],
+      [true, 'https://some-host/path?param1=value'],
+      [true, 'slack://'],
+      [true, 'mailto://'],
+      [true, 'ms-help://'],
+      [true, 'ms.help://'],
+      [true, 'ms+help://'],
+      [false, '//'],
+      [false, '123://'],
+      [false, 'abc&xzy://'],
+      [false, 'http'],
+      [false, 'https:/a'],
```

Comment:
I was about to ask you to perhaps use the URL constructor and catching errors, but actually, that one seems TOO permissive

![Screenshot 2021-03-08 at 11 57 48](https://user-images.githubusercontent.com/3097461/110312538-82a56000-8005-11eb-96f6-135112c32d77.png)

This surprised me

Reply:
Haha, we had quite a bit of a discussion about this here :) Got a bit nerd sniped.

This seems to be robust... But not very pretty!

```ts
try {
  return x === new URL(x, x).toString();
} catch {
 return false;
}
```

Reply:
Oh no! The trailing slash! Never mind

Reply:
```ts
try {
  return new URL(x).toString() === new URL(x, x).toString();
} catch {
  return false;
}
```

Was the one :p